src/main/thrift/parquet.thrift

```diff
@@ -211,6 +206,97 @@ struct Statistics {
    4: optional i64 distinct_count;
 }
 
+/** Empty structs to use as logical type annotations */
+struct StringType {}
+struct MapType {}
+struct ListType {}
+struct EnumType {}
+struct DateType {}
+
+/**
+ * Logical type to annotate a column that is always null.
+ *
+ * Sometimes when discovering the schema of existing data values are always
+ * null and the physical type is assumed.
+ */
+struct NullType {}
+
+/**
+ * Decimal logical type annotation
+ *
+ * To maintain forward-compatibility in v1, implementations using this logical
+ * type must also set scale and precision on the annotated SchemaElement.
+ */
+struct DecimalType {
+  1: required i32 scale
+  2: required i32 precision
+}
+
+/** Time units for logical types */
+struct MilliSeconds {}
+struct MicroSeconds {}
+union TimeUnit {
+  1: MilliSeconds MILLIS
+  2: MicroSeconds MICROS
+}
+
+/** Timestamp logical type annotation */
+struct TimestampType {
+  1: required bool isAdjustedToUTC
```

Comment:
We spoke about this on the Arrow sync yesterday.

In Arrow, we have 3 cases:

* No time zone in the metadata: the data is time zone naive. One can later localize to a time zone (which may alter the values, because they will be internally normalized to UTC)

* Time zone in the metadata. Whether the time zone is `'UTC'` or `'America/Los_Angeles'`, the physical values themselves are the same: changing the time zone only changes the metadata, not the values of the int64 timestamps

What is proposed here simplified this to either `isAdjustedToUTC=false` (what we currently call "no time zone" or "time zone naive" in Arrow) or `isAdjustedToUTC=true` (which covers BOTH the case that the time zone is set as UTC or some other time zone)

The problem I see here is that if a data processing system runs the query:

```sql
select hour(timestamp_field), count(*)
from my_parquet_data
group by 1
```

For timestamps with time zone, if the time zone is known then the `hour` function can be implemented to compute the hour values based on the time zone (e.g. America/Los_Angeles). But what's proposed here precludes that, you would need to do something like

```
hour_localized(timestamp_field, 'America/Los_Angeles')
...
```

Or maybe the analytics system has some means to cast to a timestamp with the additional metadata. Either way there's some awkwardness.

> This isn't done by other systems, and there is no guarantee that there is a single zone that the timestamps were converted from.

I see this concern, but if there is no consistency about the origin of the data, why not store "UTC" as the storage time zone? The fact that others may preserve a local time zone need not complicated this use case.

We can use the keyvalue metadata to preserve the time zone metadata in the Parquet file footer to at least maintain compatibility with Arrow, but it's not ideal.